app/src/main/java/com/example/game/helpers/AddressUtil.java

```diff
@@ -1,4 +1,36 @@
 package com.example.game.helpers;
 
+import android.content.Context;
+import android.location.Address;
+import android.location.Geocoder;
+import android.util.Log;
+
+import java.util.List;
+import java.util.Locale;
+
 public class AddressUtil {
+    public static String getCompleteAddressString(Context context, double LATITUDE, double LONGITUDE) {
+        String strAdd = "";
+        Geocoder geocoder = new Geocoder(context, Locale.getDefault());
+        try {
+            List<Address> addresses = geocoder.getFromLocation(LATITUDE, LONGITUDE, 1);
+            if (addresses != null) {
```

Comment:
need to check size also because you are doing get(0) which will trigger index out of bound exception for empty list.